pandas/core/groupby.py

```diff
@@ -1487,19 +1489,32 @@ def aggregate(self, values, how, axis=0):
                 raise NotImplementedError
             out_shape = (self.ngroups,) + values.shape[1:]
 
-        if is_numeric_dtype(values.dtype):
-            values = com.ensure_float(values)
-            is_numeric = True
-            out_dtype = 'f%d' % values.dtype.itemsize
-        else:
-            is_numeric = issubclass(values.dtype.type, (np.datetime64,
-                                                        np.timedelta64))
+        if _is_datetime_or_timedelta_dtype(values.dtype):
+            values = values.view('int64')
+        elif is_bool_dtype(values.dtype):
+            values = _algos.ensure_float64(values)
+
+        is_numeric = is_numeric_dtype(values.dtype)
+        upsample = self.ngroups > values.size
+
+        if is_numeric and upsample:
+            # upsample uses fillna
+            values = _algos.ensure_float64(values)
+
+        try:
+            agg_func, dtype_str = self._get_aggregate_function(how, values)
+        except NotImplementedError:
```

Comment:
haha blaming the messerger eh? 

that's what I thought.....ok, can you confirm that you are testing on this though? (e.g. that certain functions raise NotIMplementedError when you try them, e.g. mean on int64)